salt/modules/boto_iam.py

```diff
@@ -153,6 +153,297 @@ def role_exists(name, region=None, key=None, keyid=None, profile=None):
+        conn.create_user(user_name, path)
+        log.info('Created user : {0} .'.format(user_name))
+        return True
+    except boto.exception.BotoServerError as e:
+        log.debug(e)
+        msg = 'Failed to create user {0} .'
+        log.error(msg.format(user_name))
+        return False
+
+
+def get_user(user_name=None, region=None, key=None, keyid=None, profile=None):
+    '''
+    Get user information
+
+    CLI example::
+
+        salt myminion boto_iam.get_user myuser
+    '''
+    conn = _get_conn(region, key, keyid, profile)
+    try:
+        info = conn.get_user(user_name)
+        if not info:
+            return False
+        return info
+    except boto.exception.BotoServerError as e:
+        log.debug(e)
+        msg = 'Failed to get user {0} info .'
+        log.error(msg.format(user_name))
+        return False
+
+
+def create_group(group_name, path=None, region=None, key=None, keyid=None,
+                 profile=None):
+    '''
+    Create a group
+
+    CLI example::
+
+        salt myminion boto_iam.create_group group
+    '''
+    if not path:
+        path = '/'
+    if __salt__['boto_iam.get_group'](group_name=group_name, region=region, key=key, keyid=keyid, profile=profile):
+        return True
+    conn = _get_conn(region, key, keyid, profile)
+    try:
+        conn.create_group(group_name, path)
+        log.info('Created group : {0} .'.format(group_name))
+        return True
+    except boto.exception.BotoServerError as e:
+        log.debug(e)
+        msg = 'Failed to create group {0} .'
+        log.error(msg.format(group_name))
+        return False
+
+
+def get_group(group_name, marker=None, max_items=None, region=None, key=None,
+              keyid=None, profile=None):
+    '''
+    Get group information
+
+    CLI example::
+
+        salt myminion boto_iam.get_group mygroup
+    '''
+    conn = _get_conn(region, key, keyid, profile)
+    try:
+        info = conn.get_group(group_name)
+        if not info:
+            return False
+        return info
+    except boto.exception.BotoServerError as e:
+        log.debug(e)
+        msg = 'Failed to get group {0} info .'
+        log.error(msg.format(group_name))
+        return False
+
+
+def add_user_to_group(user_name, group_name, region=None, key=None, keyid=None,
+                      profile=None):
+    '''
+    Add user to group
+
+    CLI example::
+
+        salt myminion boto_iam.add_user_to_group myuser mygroup
+    '''
+    group = __salt__['boto_iam.get_group'](group_name=group_name,
+                                           region=region, key=key,
+                                           keyid=keyid, profile=profile)
+    if group:
+        for _users in group['get_group_response']['get_group_result']['users']:
+            if user_name == _users['user_name']:
+                msg = 'Username : {0} is already in group {1} .'
+                log.info(msg.format(user_name, group_name))
+                return 'Exists'
```

Comment:
Empty dict? The problem with these True/False is that you never know it is None, False or empty string or empty dict. And it is really annoying, because one of these are actually positive (True) value. And therefore this will cause a spaghetti code with tons of if/else to know which is what. :(